nnfabrik/templates/transfer/transferred_trained_model.py

```diff
@@ -0,0 +1,310 @@
+                prev_model_fn="model_fn",
+                prev_model_hash="model_hash",
+                prev_dataset_fn="dataset_fn",
+                prev_dataset_hash="dataset_hash",
+                prev_trainer_fn="trainer_fn",
+                prev_trainer_hash="trainer_hash",
+                prev_seed="seed",
+                transfer_step="transfer_step + 1",
+                prev_collapsed_history="collapsed_history",
+                _data_transfer="data_transfer",  # rename so this entry in the recipe is not used for restriction
+                prev_step="transfer_step",  # rename so this entry in the recipe is not used for restriction
+            ) * dj.U(
+                "transfer_step",  # make these attributes primary keys
+                "prev_model_fn",
+                "prev_model_hash",
+                "prev_dataset_fn",
+                "prev_dataset_hash",
+                "prev_trainer_fn",
+                "prev_trainer_hash",
+                "prev_seed",
+                "prev_collapsed_history",
+            )
+
+            # get the current transfer step
+            max_transfer_step = prev_transferred_model.fetch("transfer_step").max() if prev_transferred_model else 0
+
+            for transfer_step in range(1, max_transfer_step + 1):
+                # get the transfer recipe
+                recipe = self._transfer_recipe(transfer_step)
+                post_restr = recipe.post_restr if recipe else dj.AndList([])
+
+                # apply the recipe
+                transfer_from = prev_transferred_model * recipe
+                transfers = (
+                    self.model_table * self.dataset_table * self.trainer_table * self.seed_table * transfer_from
+                )  # combine recipe restriction with all possible training combinations
+
+                transfers = transfers * self.CollapsedHistory().proj(
+                    "collapsed_history",
+                    "prev_collapsed_history",
+                    prev_model_fn="model_fn",
+                    prev_model_hash="model_hash",
+                    prev_dataset_fn="dataset_fn",
+                    prev_dataset_hash="dataset_hash",
+                    prev_trainer_fn="trainer_fn",
+                    prev_trainer_hash="trainer_hash",
+                    prev_seed="seed",
+                )  # map previous transferred model to its collapsed history
+
+                transfers = (
+                    dj.U(
+                        "transfer_step",
+                        "model_fn",
+                        "model_hash",
+                        "dataset_fn",
+                        "dataset_hash",
+                        "trainer_fn",
+                        "trainer_hash",
+                        "seed",
+                        "collapsed_history",
+                        "data_transfer",
+                    )
+                    & transfers
+                    & post_restr  # restrict with post_rest
```

Comment:
super minor but `restrict with post_restr` - missing `r`

Reply:
I applaud your eye for detail!